prov/efa/src/efa_device.c

```diff
@@ -97,6 +97,20 @@ int efa_device_construct(struct efa_device *efa_device,
 		goto err_close;
 	}
 
+	if (-FI_ENODATA == fi_param_get_int(&rxr_prov, "use_device_rdma", &tmp_value)) {
+		if (efa_device->ibv_attr.vendor_part_id == 0xefa0 ||
+		    efa_device->ibv_attr.vendor_part_id == 0xefa1) {
+			rxr_env.use_device_rdma = 0;
+		} else {
+			if (!(efa_device->efa_attr.device_caps & EFADV_DEVICE_ATTR_CAPS_RDMA_READ)) {
+				EFA_WARN(FI_LOG_DOMAIN, "EFA device has no rdma-read capability.\n");
+				err = -FI_EINVAL;
+				goto err_close;
+			}
+			rxr_env.use_device_rdma = 1;
```

Comment:
I am not sure the action should be performed here.

IMO, this should be in `rxr_env_initialize` before `rxr_env_param_get` is called.

 There is no guarantee that this function is called before `rxr_env_initialize`.

Reply:
Should we just change https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/rxr/rxr_env.c#L45 to 1 by default? and remove the assignment here.

Reply:
Not sure I understand. where do you mean by "here"?

Reply:
I thought the `action` in your comment 

> `I am not sure the action should be performed here.` 

is the `rxr_env.use_device_rdma = 1;`  for non-0xefa0/1 device.

Reply:
>There is no guarantee that this function is called before rxr_env_initialize.

This function is called after `rxr_env_initialize`. What is the issue then?

Reply:
Another advantage of moving this code to `rxr_env_initialize` is that we will be able to include the value `rxr_env.use_device_rdma` in the message for `FI_EFA_USE_DEVICE_RDMA`. This allow user to check whether RDMA read will be used by default by running `fi_info -e`.

Reply:
I move the code to `rxr_env_initialize` in the latest revision.